### ISSUE TITLE
Deploy Bert with TRT-LLM and Triton server

### ISSUE THREAD

**Issue by chenchunhui97** [NVIDIA/TensorRT-LLM#1659]
### System Info

CPU:x86-64
GPU: a800
tensorrt-llm: v0.8.0
tensorrt-llm-backend(triton): v0.8.0
triton docker : nvcr.io/nvidia/tritonserver:24.03-trtllm-python-py3

### Who can help?

@byshiue @kaiyu

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

python3 scripts/launch_triton_server.py --world_size=1 --model_repo=all_models/inflight_batcher_llm --world_size 1

### Expected behavior

deploy the model successfully

### actual behavior

root@node5:/workspace/tensorrtllm_backend# I0524 03:33:10.919714 2517 pinned_memory_manager.cc:275] Pinned memory pool is created at '0x7f0124000000' with size 268435456
I0524 03:33:10.927952 2517 cuda_memory_manager.cc:107] CUDA memory pool is created on device 0 with size 67108864
I0524 03:33:10.947990 2517 model_lifecycle.cc:469] loading: postprocessing:1
I0524 03:33:10.948102 2517 model_lifecycle.cc:469] loading: preprocessing:1
I0524 03:33:10.948209 2517 model_lifecycle.cc:469] loading: tensorrt_llm:1
I0524 03:33:10.948292 2517 model_lifecycle.cc:469] loading: tensorrt_llm_bls:1
I0524 03:33:11.001977 2517 python_be.cc:2391] TRITONBACKEND_ModelInstanceInitialize: preprocessing_0_0 (CPU device 0)
I0524 03:33:11.002041 2517 python_be.cc:2391] TRITONBACKEND_ModelInstanceInitialize: postprocessing_0_0 (CPU device 0)
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.
I0524 03:33:12.154632 2517 model_lifecycle.cc:835] successfully loaded 'preprocessing'
I0524 03:33:12.217227 2517 model_lifecycle.cc:835] successfully loaded 'postprocessing'
I0524 03:33:12.863387 2517 python_be.cc:2391] TRITONBACKEND_ModelInstanceInitialize: tensorrt_llm_bls_0_0 (CPU device 0)
[TensorRT-LLM][WARNING] gpu_device_ids is not specified, will be automatically set
[TensorRT-LLM][WARNING] batch_scheduler_policy parameter was not found or is invalid (must be max_utilization or guaranteed_no_evict)
[TensorRT-LLM][WARNING] enable_chunked_context is not specified, will be set to false.
[TensorRT-LLM][WARNING] enable_trt_overlap is not specified, will be set to false
[TensorRT-LLM][WARNING] normalize_log_probs is not specified, will be set to true
[TensorRT-LLM][WARNING] Parameter version cannot be read from json:
[TensorRT-LLM][WARNING] [json.exception.out_of_range.403] key 'version' not found
[TensorRT-LLM][INFO] No engine version found in the config file, assuming engine(s) built by old builder API.
[TensorRT-LLM][WARNING] Parameter pipeline_parallel cannot be read from json:
[TensorRT-LLM][WARNING] [json.exception.out_of_range.403] key 'pipeline_parallel' not found
E0524 03:33:12.868443 2517 backend_model.cc:691] ERROR: Failed to create instance: unexpected error when creating modelInstanceState: [json.exception.out_of_range.403] key 'num_layers' not found
E0524 03:33:12.868582 2517 model_lifecycle.cc:638] failed to load 'tensorrt_llm' version 1: Internal: unexpected error when creating modelInstanceState: [json.exception.out_of_range.403] key 'num_layers' not found
I0524 03:33:12.868608 2517 model_lifecycle.cc:773] failed to load 'tensorrt_llm'
I0524 03:33:13.129702 2517 model_lifecycle.cc:835] successfully loaded 'tensorrt_llm_bls'
E0524 03:33:13.129882 2517 model_repository_manager.cc:579] Invalid argument: ensemble 'ensemble' depends on 'tensorrt_llm' which has no loaded version. Model 'tensorrt_llm' loading failed with error: version 1 is at UNAVAILABLE state: Internal: unexpected error when creating modelInstanceState: [json.exception.out_of_range.403] key 'num_layers' not found;
I0524 03:33:13.130066 2517 server.cc:607]
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I0524 03:33:13.130212 2517 server.cc:634]
+-------------+-------------------------------------------------------+-------------------------------------------------------+
| Backend     | Path                                                  | Config                                                |
+-------------+-------------------------------------------------------+-------------------------------------------------------+
| python      | /opt/tritonserver/backends/python/libtriton_python.so | {"cmdline":{"auto-complete-config":"false","backend-d |
|             |                                                       | irectory":"/opt/tritonserver/backends","min-compute-c |
|             |                                                       | apability":"6.000000","shm-region-prefix-name":"prefi |
|             |                                                       | x0_","default-max-batch-size":"4"}}                   |
| tensorrtllm | /opt/tritonserver/backends/tensorrtllm/libtriton_tens | {"cmdline":{"auto-complete-config":"false","backend-d |
|             | orrtllm.so                                            | irectory":"/opt/tritonserver/backends","min-compute-c |
|             |                                                       | apability":"6.000000","default-max-batch-size":"4"}}  |
|             |                                                       |                                                       |
+-------------+-------------------------------------------------------+-------------------------------------------------------+

I0524 03:33:13.130393 2517 server.cc:677]
+------------------+---------+------------------------------------------------------------------------------------------------+
| Model            | Version | Status                                                                                         |
+------------------+---------+------------------------------------------------------------------------------------------------+
| postprocessing   | 1       | READY                                                                                          |
| preprocessing    | 1       | READY                                                                                          |
| tensorrt_llm     | 1       | UNAVAILABLE: Internal: unexpected error when creating modelInstanceState: [json.exception.out_ |
|                  |         | of_range.403] key 'num_layers' not found                                                       |
| tensorrt_llm_bls | 1       | READY                                                                                          |
+------------------+---------+------------------------------------------------------------------------------------------------+

I0524 03:33:16.524488 2517 metrics.cc:877] Collecting metrics for GPU 0: NVIDIA A800 80GB PCIe
I0524 03:33:16.537087 2517 metrics.cc:770] Collecting CPU metrics
I0524 03:33:16.537474 2517 tritonserver.cc:2538]
+----------------------------------+--------------------------------------------------------------------------------------------+
| Option                           | Value                                                                                      |
+----------------------------------+--------------------------------------------------------------------------------------------+
| server_id                        | triton                                                                                     |
| server_version                   | 2.44.0                                                                                     |
| server_extensions                | classification sequence model_repository model_repository(unload_dependents) schedule_poli |
|                                  | cy model_configuration system_shared_memory cuda_shared_memory binary_tensor_data paramete |
|                                  | rs statistics trace logging                                                                |
| model_repository_path[0]         | all_models/inflight_batcher_llm                                                            |
| model_control_mode               | MODE_NONE                                                                                  |
| strict_model_config              | 1                                                                                          |
| rate_limit                       | OFF                                                                                        |
| pinned_memory_pool_byte_size     | 268435456                                                                                  |
| cuda_memory_pool_byte_size{0}    | 67108864                                                                                   |
| min_supported_compute_capability | 6.0                                                                                        |
| strict_readiness                 | 1                                                                                          |
| exit_timeout                     | 30                                                                                         |
| cache_enabled                    | 0                                                                                          |
+----------------------------------+--------------------------------------------------------------------------------------------+

I0524 03:33:16.537498 2517 server.cc:307] Waiting for in-flight requests to complete.
I0524 03:33:16.537527 2517 server.cc:323] Timeout 30: Found 0 model versions that have in-flight inferences
I0524 03:33:16.538828 2517 server.cc:338] All models are stopped, unloading models
I0524 03:33:16.538853 2517 server.cc:347] Timeout 30: Found 3 live models and 0 in-flight non-inference requests
I0524 03:33:17.538960 2517 server.cc:347] Timeout 29: Found 3 live models and 0 in-flight non-inference requests
Cleaning up...
Cleaning up...
Cleaning up...
I0524 03:33:18.539108 2517 server.cc:347] Timeout 28: Found 3 live models and 0 in-flight non-inference requests
I0524 03:33:18.543341 2517 model_lifecycle.cc:620] successfully unloaded 'tensorrt_llm_bls' version 1
I0524 03:33:19.033556 2517 model_lifecycle.cc:620] successfully unloaded 'preprocessing' version 1
I0524 03:33:19.446490 2517 model_lifecycle.cc:620] successfully unloaded 'postprocessing' version 1
I0524 03:33:19.539250 2517 server.cc:347] Timeout 27: Found 0 live models and 0 in-flight non-inference requests
error: creating server: Internal - failed to load all models
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[8115,1],0]
  Exit code:    1
--------------------------------------------------------------------------

### additional notes

model config:
{
  "_name_or_path": "/storage/nfs2/ModelHub/embedding/BAAI/reranker/bge-reranker-large",
  "architectures": [
    "XLMRobertaForSequenceClassification"
  ],
  "attention_probs_dropout_prob": 0.1,
  "bos_token_id": 0,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 1024,
  "id2label": {
    "0": "LABEL_0"
  },
  "initializer_range": 0.02,
  "intermediate_size": 4096,
  "label2id": {
    "LABEL_0": 0
  },
  "layer_norm_eps": 1e-05,
  "max_position_embeddings": 514,
  "model_type": "xlm-roberta",
  "num_attention_heads": 16,
  "num_hidden_layers": 24,
  "output_past": true,
  "pad_token_id": 1,
  "position_embedding_type": "absolute",
  "torch_dtype": "float32",
  "transformers_version": "4.37.2",
  "type_vocab_size": 1,
  "use_cache": true,
  "vocab_size": 250002,
  "num_layers":24
}

**Comment by chenchunhui97** [comment]
key info:

E0524 03:33:12.868443 2517 backend_model.cc:691] ERROR: Failed to create instance: unexpected error when creating modelInstanceState: [json.exception.out_of_range.403] key 'num_layers' not found
E0524 03:33:12.868582 2517 model_lifecycle.cc:638] failed to load 'tensorrt_llm' version 1: Internal: unexpected error when creating modelInstanceState: [json.exception.out_of_range.403] key 'num_layers' not found
I0524 03:33:12.868608 2517 model_lifecycle.cc:773] failed to load 'tensorrt_llm'
I0524 03:33:13.129702 2517 model_lifecycle.cc:835] successfully loaded 'tensorrt_llm_bls'
E0524 03:33:13.129882 2517 model_repository_manager.cc:579] Invalid argument: ensemble 'ensemble' depends on 'tensorrt_llm' which has no loaded version. Model 'tensorrt_llm' loading failed with error: version 1 is at UNAVAILABLE state: Internal: unexpected error when creating modelInstanceState: [json.exception.out_of_range.403] key 'num_layers' not found;
I0524 03:33:13.130066 2517 server.cc:607]

1. can I deploy BERT(encoder-only model) with triton?
2. Should I install tensorrt-llm again in the triton container?
3. I have specified `num_layers` in config but why it raise key 'num_layers' not found?

**Comment by byshiue** [comment]
Currently, we don't support running BERT model by TRT-LLM on triton.

You can try using TRT directly.
